apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/trace/EntrySpan.java

```diff
@@ -96,6 +96,9 @@ public AbstractTracingSpan setComponent(String componentName) {
 
     @Override
     public AbstractTracingSpan setOperationName(String operationName) {
+        if (isInAsyncMode) {
+            throw new RuntimeException("Span is in async mode, Should use setOperationNameInAsync method");
```

Comment:
This method is called by `#setOperationNameInAsync`. I am pretty sure this will be triggered :)

Reply:
Don't throw an exception, it caused too much.

Reply:
Isn't it necessary to add ``setOperationNameInAsync``,  judging from this place is async mode, and modify ``operationName``,Isn't this better, only one api

Reply:
Make sense. Let's keep API unchanged. And revert all other changes.